x-pack/filebeat/input/gcs/state.go

```diff
@@ -62,12 +71,28 @@ func (s *state) save(name string, lastModifiedOn *time.Time) {
 	s.mu.Unlock()
 }
 
+// setRootArray, sets boolean true for objects that have their roots defined as an array type
+func (s *state) setRootArray(name string) {
+	s.cp.IsRootArray[name] = true
+}
+
+// savePartial, partially saves/updates the current state for cursor checkpoint
+func (s *state) savePartial(name string, offset int64) {
+	s.mu.Lock()
+	s.cp.PartiallyProcessed[name] = offset
+	s.mu.Unlock()
```

Comment:
Why does this use locking when the rest of the `state` setters don't and reads of `s.cp.PartiallyProcesses` are not protected?

Reply:
this is not required, since at any given point , only 1 go routine will work on 1 job, and the job acquired by 1 routine cannot be acquired by others in the same passthrough. Will remove this.